js/components/frame.js

```diff
@@ -802,7 +810,11 @@ class Frame extends ImmutableComponent {
       const isError = this.props.aboutDetails && this.props.aboutDetails.get('errorCode')
       if (!this.props.isPrivate && this.props.provisionalLocation === this.props.location && (protocol === 'http:' || protocol === 'https:') && !isError && savePage) {
         // Register the site for recent history for navigation bar
-        appActions.addSite(siteUtil.getDetailFromFrame(this.frame))
+        const addSiteDebounced = debounce((frame) => {
```

Comment:
We were debouncing this so that we can call it multiple times. As the different events come in.
So you need to move `addSiteDebounced` above at the top of the file.  Then use it like you do below. But also use it in the other events as things come in like the favicon and the themeColor event.